Code/Framework/AzCore/AzCore/Utils/Utils.cpp

```diff
@@ -241,6 +240,42 @@ namespace AZ::Utils
         return {};
     }
 
+    AZ::IO::FixedMaxPathString GetProjectLogPath(AZ::SettingsRegistryInterface* settingsRegistry)
+    {
+        if (settingsRegistry == nullptr)
+        {
+            settingsRegistry = AZ::SettingsRegistry::Get();
+        }
+
+        if (settingsRegistry != nullptr)
+        {
+            if (AZ::IO::FixedMaxPathString settingsValue;
+                settingsRegistry->Get(settingsValue, AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectLogPath))
+            {
+                return settingsValue;
+            }
+        }
+        return {};
+    }
+
+    AZ::IO::FixedMaxPathString GetProjectProductPathForPlatform(AZ::SettingsRegistryInterface* settingsRegistry)
+    {
+        if (settingsRegistry == nullptr)
+        {
+            settingsRegistry = AZ::SettingsRegistry::Get();
+        }
+
+        if (settingsRegistry != nullptr)
+        {
+            if (AZ::IO::FixedMaxPathString settingsValue;
+                settingsRegistry->Get(settingsValue, AZ::SettingsRegistryMergeUtils::FilePathKey_CacheRootFolder))
+            {
+                return settingsValue;
+            }
+        }
+        return {};
+    }
```

Comment:
we're getting enough of these similar `GetXYZPath` util functions where the only difference seems to be the key - I wonder if we should have a 
`AZ::IO::FixedMaxPathString GetPathFromKey(settingsRegistry, key)` function to DRY things up a bit.

Reply:
That would decrease the number of lines of code a bit